openmc/tallies.py

```diff
@@ -1620,108 +1660,129 @@ def _align_tally_data(self, other):
+            other._mean = np.tile(other.mean, (1, self.num_nuclides, 1))
+            other._std_dev = np.tile(other.std_dev, (1, self.num_nuclides, 1))
+
+        # Add nuclides to each tally such that each tally contains the complete
+        # set of nuclides necessary to perform an entrywise product. New nuclides
+        # added to a tally will have all their scores set to zero.
+        else:
+
+            # Get the set of nuclides that each tally is missing
+            other_missing_nuclides = set(self.nuclides).difference(set(other.nuclides))
+            self_missing_nuclides = set(other.nuclides).difference(set(self.nuclides))
+
+            # Add nuclides present in self but not in other to other
+            for nuclide in other_missing_nuclides:
+                other._mean = np.insert(other.mean, other.num_nuclides, 0, axis=1)
+                other._std_dev = np.insert(other.std_dev, other.num_nuclides, 0, axis=1)
+                other.add_nuclide(nuclide)
+
+            # Add nuclides present in other but not in self to self
+            for nuclide in self_missing_nuclides:
+                self._mean = np.insert(self.mean, self.num_nuclides, 0, axis=1)
+                self._std_dev = np.insert(self.std_dev, self.num_nuclides, 0, axis=1)
+                self.add_nuclide(nuclide)
+
+            # Align other nuclides with self nuclides
+            for i, nuclide in enumerate(self.nuclides):
+                other_index = other.get_nuclide_index(nuclide)
+
+                # If necessary, swap other nuclide
+                if other_index != i:
+                    other._swap_nuclides(nuclide, other.nuclides[i])
+
+        # Repeat and tile the data by score in preparation for performing
+        # the tensor product across scores.
+        if score_product == 'tensor':
+            self._mean = np.repeat(self.mean, other.num_score_bins, axis=2)
+            self._std_dev = np.repeat(self.std_dev, other.num_score_bins, axis=2)
+            other._mean = np.tile(other.mean, (1, 1, self.num_score_bins))
+            other._std_dev = np.tile(other.std_dev, (1, 1, self.num_score_bins))
+
+        # Add scores to each tally such that each tally contains the complete set
+        # of scores necessary to perform an entrywise product. New scores added
+        # to a tally will be set to zero.
+        else:
+
+            # Get the set of scores that each tally is missing
+            other_missing_scores = set(self.scores).difference(set(other.scores))
+            self_missing_scores = set(other.scores).difference(set(self.scores))
+
+            # Add scores present in self but not in other to other
+            for score in other_missing_scores:
+                other._mean = np.insert(other.mean, other.num_score_bins, 0, axis=2)
+                other._std_dev = np.insert(other.std_dev, other.num_score_bins, 0, axis=2)
+                other.add_score(score)
```

Comment:
If we do indeed need to keep `num_score_bins` around, why is it not incremented for `other` here as it is for `self` below?